src/wallet/CarbonWallet.ts

```diff
@@ -758,20 +730,11 @@ export class CarbonWallet {
   }
 
   public async reloadTxFees() {
-    const queryClient = this.getQueryClient();
-    const { msgGasCosts } = await queryClient.fee.MsgGasCostAll({});
-    this.txGasCosts = msgGasCosts.reduce((result, item) => {
-      result[item.msgType] = bnOrZero(item.gasCost);
-      return result;
-    }, {} as SimpleMap<BigNumber>);
-
+    const txGasPrices = this.gasFee?.txGasPrices ?? {}
+    const queryClient = this.getQueryClient()
     const { minGasPrices } = await queryClient.fee.MinGasPriceAll({});
```

Comment:
Might want to move L733 - L734 into the if statement on L737 so that this query does not run more than necessary.
i.e.
```typescript
if (!txGasPrices[this.defaultFeeDenom]) {
  const queryClient = this.getQueryClient()
  const { minGasPrices } = await queryClient.fee.MinGasPriceAll({});
  ...
}
```